app/uk/gov/hmrc/gform/validation/ValidationService.scala

```diff
@@ -434,44 +465,49 @@ class ComponentsValidator(
       case (FractionalShape(_, whole, _, _), _, _) if filterCommas(whole).size > maxWhole =>
-        getError(fieldValue, s"number must be at most $maxWhole whole digits")
+        getError(fieldValue, s"must be at most $maxWhole whole digits")
       case (FractionalShape(_, _, _, fractional), 0, _) if fractional.size > 0 =>
         getError(fieldValue, "must be a whole number")
       case (FractionalShape(_, _, _, fractional), _, _) if fractional.size > maxFractional =>
-        getError(fieldValue, s"decimal fraction must be at most $maxFractional digits")
-      case (FractionalShape("-", _, _, _), _, true) => getError(fieldValue, "must be a positive number")
-      case (FractionalShape(_, _, _, _), _, _)      => ().valid
-      case (_, 0, true)                             => getError(fieldValue, "must be a positive whole number")
-      case (_, _, true)                             => getError(fieldValue, "must be a positive number")
-      case (_, 0, false)                            => getError(fieldValue, "must be a whole number")
-      case _                                        => getError(fieldValue, "must be a number")
+        getError(fieldValue, s"must be at most $maxFractional digits")
+      case (FractionalShape("-", _, _, _), _, true) =>
+        getError(fieldValue, "must be a positive number")
+      case (FractionalShape(_, _, _, _), _, _) => ().valid
+      case (_, 0, true)                        => getError(fieldValue, "must be a positive whole number")
+      case (_, _, true)                        => getError(fieldValue, "must be a positive number")
+      case (_, 0, false)                       => getError(fieldValue, "must be a whole number")
+      case _                                   => getError(fieldValue, "must be a number")
     }
   }
 
-  private def validateRequired(fieldValue: FormComponent, fieldId: FormComponentId)(xs: Seq[String]): ValidatedType =
+  private def validateRequired(fieldValue: FormComponent, fieldId: FormComponentId, errorPrefix: Option[String] = None)(
+    xs: Seq[String]): ValidatedType =
     xs.filterNot(_.isEmpty()) match {
-      case Nil           => Map(fieldId -> errors(fieldValue, "must be entered")).invalid
+      case Nil =>
+        Map(fieldId -> errors(fieldValue, s"${errorPrefix.getOrElse(messagePrefix(fieldValue))} must be entered")).invalid
       case value :: Nil  => ().valid
       case value :: rest => ().valid // we don't support multiple values yet
     }
 
   private def validateForbidden(fieldValue: FormComponent, fieldId: FormComponentId)(xs: Seq[String]): ValidatedType =
     xs.filterNot(_.isEmpty()) match {
-      case Nil          => ().valid
-      case value :: Nil => Map(fieldId -> errors(fieldValue, "must not be entered")).invalid
+      case Nil => ().valid
+      case value :: Nil =>
+        Map(fieldId -> errors(fieldValue, s"${messagePrefix(fieldValue)} must not be entered")).invalid
```

Comment:
We still have some repetition of `${messagePrefix(fieldValue)}` in calls to `errors`. It would be nice to get rid of it. 
We don't use `${messagePrefix(fieldValue)}` in every call to `errors` though, so for these cases we can have different method, which do not append any prefix.

New API may look like this:

```
private def errors(fieldValue: FormComponent, defaultErr: String): Set[String] =
    Set(fieldValue.errorMessage.getOrElse(messagePrefix(fieldValue) + " " + defaultErr))

  private def errorsNoPrefix(fieldValue: FormComponent, defaultErr: String): Set[String] =
    Set(fieldValue.errorMessage.getOrElse(defaultErr))

  private def getError(fieldValue: FormComponent, defaultMessage: String) =
    Map(fieldValue.id -> errors(fieldValue, defaultMessage)).invalid
```

Reply:
I will update the API and try it out - main use case for no prefix is the address line errors (the example above should have been using messagePrefix and was updated to do so in later commit)

Reply:
Example above will keep using prefix with that new API.

Reply:
This is now done - I did not need errorsNoPrefix() and have it all working ok with passing tests.

Reply:
That is even better outcome than I've expected, well done.